src/shared/userdb.c

```diff
@@ -1669,7 +1669,7 @@ static void discover_membership_dropins(UserDBIterator *i, UserDBFlags flags) {
                         &i->dropins,
                         ".membership",
                         NULL,
-                        CONF_FILES_REGULAR|CONF_FILES_BASENAME|CONF_FILES_FILTER_MASKED,
+                        CONF_FILES_REGULAR|CONF_FILES_BASENAME,
```

Comment:
hmm, it's not that simple, we check for /dev/null symlink masking before we enforce CONF_FILES_REGULAR. I think we should retain the ability to mask via such a symlink. 

Maybe we should split CONF_FILES_FILTER_MASKED into CONF_FILES_FILTER_MASKED_BY_SYMLINK and CONF_FILES_FILTER_MASKED_BY_EMPTY and then make CONF_FILES_FILTER_MASKED into an OR combination of both. Then we could use CONF_FILES_FILTER_MASKED_BY_SYMLINK here, which we want, but leave CONF_FILES_FILTER_MASKD_BY_EMPTY out.

Reply:
Ah, you mean, the case that a file that have higher priority is a symlink to /dev/null, and lower priority has an empty file.

Reply:
OK. I will split the MASKED flag into two.